Festispec.App/ViewModels/FormOverviewViewModel.cs

```diff
@@ -7,40 +8,82 @@
 	{
-        private IFormsRepository formRepository;
+        private readonly IFormsRepository _formRepository;
 
         public ObservableCollection<FormViewModel> Forms { get;}
 		public ObservableCollection<FormViewModel> Templates { get; }
-		public FormViewModel SelectedForm { get; set; }
+        public string NewFormText { get; set; }
+        public FormViewModel SelectedForm { get; set; }
 		public FormViewModel SelectedTemplate { get; set; }
 		public RelayCommand EditCommand { get; private set; }
 		public RelayCommand RemoveCommand { get; private set; }
 		public RelayCommand CreateCommand { get; private set; }
+		public RelayCommand CreateBasedOnTemplateCommand { get; private set; }
+
+
         public FormOverviewViewModel()
         {
-            formRepository = new FormsTestRepository();
-            Forms = new ObservableCollection<FormViewModel>(formRepository.GetAll().Select(o => new FormViewModel(o)));
+            _formRepository = new FormsTestRepository();
+            Forms = new ObservableCollection<FormViewModel>(_formRepository.GetAll().Where(o => !o.IsTemplate).Select(o => new FormViewModel(o)));
+            Templates = new ObservableCollection<FormViewModel>(_formRepository.GetAll().Where(o => o.IsTemplate).Select(o => new FormViewModel(o)));
+			EditCommand = new RelayCommand(Edit, CanEditOrRemove);
+			RemoveCommand = new RelayCommand(Remove, CanEditOrRemove);
+			CreateCommand = new RelayCommand(Create, CanCreate);
+			CreateBasedOnTemplateCommand = new RelayCommand(CreateBasedOnTemplate, CanCreateBasedOnTemplate);
         }
-		public void CanEditOrRemove()
+
+		internal override bool CanEditOrRemove()
 		{
+            return SelectedForm != null && Forms.Contains(SelectedForm);
 		}
 
-		public void Create()
+		internal override bool CanCreate()
 		{
+			return !string.IsNullOrEmpty(NewFormText);
 		}
 
-		public void CreateBasedOnTemplate()
+		private bool CanCreateBasedOnTemplate()
 		{
+			return CanCreate() && SelectedTemplate != null;
 		}
 
-		public void Edit()
+		internal override void Create()
+		{
+            Form f = new Form()
+            {
+                IsTemplate = false,
+                Name = NewFormText
+            };
+            NewFormText = null;
+            _formRepository.Add(f);
+            Forms.Add(new FormViewModel(f));
+		}
+
+        private void CreateBasedOnTemplate()
+        {
+            if (SelectedTemplate == null) { return; }
+            Form f = new Form()
+            {
+                IsTemplate = false,
+                Name = NewFormText
+            };
+            foreach (QuestionViewModel q in SelectedTemplate.Questions) f.Question.Add(q.ToModel());
+			NewFormText = string.Empty;
+            _formRepository.Add(f);
+            Forms.Add(new FormViewModel(f));
+        }
+
+		internal override void Edit()
 		{
+            throw new NotImplementedException();
 		}
 
-		public void Remove()
+		internal override void Remove()
```

Comment:
Het heeft weinig toegevoegde waarde om internals te gebruiken, snap de keuze niet omdat we toch binnen de zelfde assembly werken.